### PR TITLE
Convert functions to templates

### PR DESCRIPTION
**Pull request by m4thieulavoie**
# Pull Request

## 📖 Description

This PR leverages `FoundationElementTemplate` instead of functions in component templates and styles. It should help with the typings by being more accurate. More details can be found [in the issue](https://github.com/microsoft/fast/issues/5347)

Fixes https://github.com/microsoft/fast/issues/5347

### General

- [x] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

**Comment by m4thieulavoie** [comment]
@radium-v I can't request review in the PR, but I figured you might wanna have a look :+1:

File: packages/web-components/fast-components/src/accordion-item/accordion-item.styles.ts [hunk 1]
```diff
@@ -27,13 +28,10 @@ import { heightNumber } from "../styles/size";
...
-    context: ElementDefinitionContext,
-    definition: AccordionItemOptions
-) => ElementStyles = (
-    context: ElementDefinitionContext,
-    definition: AccordionItemOptions
-) =>
+export const accordionItemStyles: FoundationElementTemplate<
+    ElementStyles,
+    AccordionItemOptions
+> = (context: ElementDefinitionContext, definition: AccordionItemOptions) =>
```
Comment (radium-v) [inline packages/web-components/fast-components/src/accordion-item/accordion-item.styles.ts:34]: The parameter types are provided by the `FoundationElementTemplate` generics, so `context` and `definition` don't need to be explicitly typed:

```suggestion
export const accordionItemStyles: FoundationElementTemplate<
    ElementStyles,
    AccordionItemOptions
> = (context, definition) =>
```

This also allows us to remove the `ElementDefinitionContext` and `FoundationElementDefinition` type imports where present.

File: packages/web-components/fast-components/src/accordion/accordion.styles.ts [hunk 2]
```diff
@@ -17,10 +18,7 @@ import {
...
  * @public
  */
-export const accordionStyles: (
-    context: ElementDefinitionContext,
-    definition: FoundationElementDefinition
-) => ElementStyles = (
+export const accordionStyles: FoundationElementTemplate<ElementStyles> = (
     context: ElementDefinitionContext,
     definition: FoundationElementDefinition
 ) =>
```
Comment (radium-v) [inline packages/web-components/fast-components/src/accordion/accordion.styles.ts:24]:
```suggestion
export const accordionStyles: FoundationElementTemplate<ElementStyles> = (
    context,
    definition
) =>
```

**Review by radium-v — CHANGES_REQUESTED** [review]

File: packages/web-components/fast-components/src/accordion/accordion.styles.ts (hunk 2 above)
Comment (m4thieulavoie) [reply packages/web-components/fast-components/src/accordion/accordion.styles.ts]: Should be done!

**Review by chrisdholt — APPROVED** [review]
@radium-v can you take another look at this when you have a moment?

**Review by radium-v — APPROVED** [review]

**Pull request merged**